presto-spark-base/src/main/java/com/facebook/presto/spark/execution/PrestoSparkTaskExecutorFactory.java

```diff
@@ -735,17 +745,58 @@ public boolean hasNext()
             throw new RuntimeException(failure);
         }
 
+        private static SerializedTaskInfo createSerializedTaskInfo(
+                TaskContext taskContext,
+                TaskStateMachine taskStateMachine,
+                UUID taskInstanceId,
+                OutputBufferType outputBufferType,
+                PrestoSparkOutputBuffer<?> outputBuffer,
+                Codec<TaskInfo> taskInfoCodec,
+                DataSize maxTaskInfoSize)
+        {
+            SerializedTaskInfo serializedTaskInfo = null;
+            for (SerializedTaskInfoCompactionLevel compactionLevel : SerializedTaskInfoCompactionLevel.values()) {
```

Comment:
I assume we do this infrequently and so this is adequate looping through the different levels is adequate?

Reply:
Yeah, we don't expect the stats to be dropped in a common case. But generally it should be cheap to recreate / serialize the object.

Reply:
I just mean it won't repeat since we only send the task updates once per task at the end?

Reply:
Yeah